overlays/bootstrap.nix

```diff
@@ -615,7 +618,9 @@ in {
                 extra-passthru = { buildGHC = final.buildPackages.haskell-nix.compiler.ghc922; };
 
                 bootPkgs = bootPkgs // {
-                  ghc = final.buildPackages.buildPackages.haskell-nix.compiler.ghc8107;
+                  ghc = if final.stdenv.buildPlatform.config == final.stdenv.targetPlatform.config
+                    then final.buildPackages.buildPackages.haskell-nix.compiler.ghc8107
+                    else final.buildPackages.buildPackages.haskell-nix.compiler.ghc922;
```

Comment:
Why? Can we add a comment as to why the bootGHC is 8107 if they are equal and 922 if not?

Reply:
I think I worked out what this was.  When I first tried to get Hadrian working 9.2.2 was the current GHC version and I had trouble with the cross compiler and hoped using the native compiler to build it would help.  When started working on getting 9.4 working instead I forgot to revert this.  It is reverted now.